pkg/server/service.go

```diff
@@ -97,7 +106,16 @@ func (s *Server) CreateEntry(ctx context.Context, req *pb.CreateEntryRequest) (*
 			slog.Warn("failed validating dsse request", "error", err.Error())
 			return nil, status.Errorf(codes.InvalidArgument, "invalid dsse request")
 		}
-		serialized, err = protojson.Marshal(entry)
+		dsseEntry := &pb.Entry{
+			Kind:       "dsse",
+			ApiVersion: "0.0.2",
+			Spec: &pb.Spec{
+				Spec: &pb.Spec_DsseV0_0_2{
+					DsseV0_0_2: entry,
+				},
+			},
+		}
+		serialized, err = protojson.Marshal(dsseEntry)
 		if err != nil {
 			slog.Warn("failed marshaling dsse request", "error", err.Error())
 			return nil, status.Errorf(codes.InvalidArgument, "invalid dsse request")
```

Comment:
We should also update line 144 to reuse the kind and version from the Entry, we can remove `types.GetKindVersion` then.